esphome/components/whirlpool/whirlpool.h

```diff
@@ -47,6 +47,8 @@ class WhirlpoolClimate : public climate_ir::ClimateIR {
   void transmit_state() override;
   /// Handle received IR Buffer
   bool on_receive(remote_base::RemoteReceiveData data) override;
+  /// Set the time of the last transmission.
+  int32_t last_transmit_time_;
```

Comment:
Sorry, I led you astray there, this should have an initialiser, just not `millis()`. Initiialising to `{}` will make it zero, so guaranteed to be in the past.

```suggestion
  int32_t last_transmit_time_{};
```